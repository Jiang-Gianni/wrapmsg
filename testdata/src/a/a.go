package a

import (
	"fmt"
)

func f() error {
	// call same package
	if err := g(); err != nil {
		return fmt.Errorf("hoge: %w", err) // want `the error-wrapping message should be "g: %w"`
	}
	return nil
}

func g() error {
	return nil
}
