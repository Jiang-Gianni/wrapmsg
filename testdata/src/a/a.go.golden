package a

import (
	"fmt"
)

func f() error {
	// non-error
	_ = fmt.Errorf("new error")
	_ = fmt.Errorf("new error with format: %d", 10)
	var msg string
	_ = fmt.Errorf(msg)

	// call same package
	if err := g(); err != nil {
		return fmt.Errorf("g: %w", err) // want `the error-wrapping message should be "g: %w"`
	}
	if err := g(); err != nil {
		return fmt.Errorf("g: %w", err)
	}
	return nil
}

func g() error {
	return nil
}
